plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/TemplateViewerFilter.java

```diff
@@ -38,15 +38,19 @@ public boolean select(Viewer viewer, Object parentElement, Object element) {
 		if(!(element instanceof IResource)) {
 			return false;
 		}
+		final String text = filterText.getText();
+		if(StringUtils.isBlank(text)) {
+			return true;
+		}
 		IResource resource = (IResource) element;
-		if(StringUtils.isBlank(filterText.getText())) {
+		if(resource.getName().contains(text)) {
```

Comment:
I would think that when there's no name then the filter would not match the given criteria in regard of the name? That's at least how I handled matching in the list of quickstarts/cartridges in v2 app wizard.